### DISCUSSION TITLE
Help with random crash (at JavascriptCallbackProxy.Finalize)

### DISCUSSION THREAD

**Discussion by eadan-y** [cefsharp/CefSharp#3555] [Q&A]
Hey, I get a random crash at my code. I can't reproduce it.
 here is the stack trace:
`System.NullReferenceException: Object reference not set to an instance of an object.\r\n   at 
CefSharp.Internals.ClientAdapter.GetBrowserWrapper(ClientAdapter* , Int32 browserId)\r\n   at 
CefSharp.ManagedCefBrowserAdapter.GetBrowser(Int32 browserId)\r\n   at 
CefSharp.Internals.JavascriptCallbackProxy.GetBrowser()\r\n   at 
CefSharp.Internals.JavascriptCallbackProxy.!JavascriptCallbackProxy()\r\n   at 
CefSharp.Internals.JavascriptCallbackProxy.Dispose(Boolean A_0)\r\n   at 
CefSharp.Internals.JavascriptCallbackProxy.Finalize()"}
`

My best guess was that something in the GC phase depends on timings (I currently don't call "Dispose" for anything on myself).
I can see this commit from 3 days ago:
https://github.com/cefsharp/CefSharp/commit/1850f3e0aff0a029b3492df81bca7d58fe613bce#diff-d3fca5475de320e364b0d3c39e37042ff5a271d60a9ce1dfb1ba08f9d0fa2b49
(by @amaitland )

Should that fix my issue?

**Comment by amaitland** [comment]
> My best guess was that something in the GC phase depends on timings (I currently don't call "Dispose" for anything on myself).

Disposing of the callbacks is recommended and should be done where possible. 

> Should that fix my issue?

Hopefully, please test with version 90 and confirm.
